### ISSUE TITLE
HazelcastClient JavaDoc should use "unicast client" instead of "dumb client"

### ISSUE THREAD

**Issue by jerrinot** [hazelcast/hazelcast#14213]
See: https://github.com/jerrinot/hazelcast/blob/3438e05a2e8e8a1ac21b9f92798d55ae3b9a9f78/hazelcast-client/src/main/java/com/hazelcast/client/HazelcastClient.java#L33
```
 * <h1>Smart vs dumb clients</h1>
 * Hazelcast Client enables you to do all Hazelcast operations without being a member of the cluster. Clients can be:
 * <ol>
 * <li>smart: this means that they immediately can send an operation like map.get(key) to the member that owns that
 * specific key.
 * </li>
 * <li>
 * dumb: it will connect to a random member in the cluster and send requests to this member. This member then needs
 * to send the request to the correct member.
```

**Comment by tezc** [comment]
We currently use "dummy client" in the documentation.

https://docs.hazelcast.org//docs/latest-dev/manual/html-single/#java-client-operation-modes

**Comment by jerrinot** [comment]
@sancar: no, I was just having a blackout:) of course it should be `unisocket`. thanks for spotting this. 
@tezc: well spotted, I believe we should consistently use the `unisocket` term at all places.

**Comment by sancar** [comment]
I will fix the javadocs with this pr. I have also opened an issue in the reference manual to keep track of the issue. 
https://github.com/hazelcast/hazelcast-reference-manual/issues/620

**Comment by sancar** [comment]
fixed by f02626a4c0e058ca29b764349f0ca959902c2a5b
